### ISSUE TITLE
Convert Flutter unit to millimeter

### ISSUE THREAD

**Issue by tixngo-mobile-github** [flutter/flutter#79252]
Hi Guys,
I am using Flutter to develop an app that have a ruler inside
By definition it should be correct:
Flutter’s unit is DP or dip (aka density independent px https://developer.android.com/training/multiscreen/screendensities#TaskUseDP ). it's independent from the screen pixel ratio.

Converting mm>dp or dp>mm is similar to mm>inch/inch>mm.
The relationship between them is fairly simple : 1 inch = 25.4 mm = 160.0 DP Which means 1 mm = 6.299 DP.

So I have a code to convert like this:
```
double fromMMToFlutterUnit(double mm) {
  return mm * 6.299;
}
```
it should be correct but in some big screen like iPad or tablet it become wrong.

is there any corrected way to convert flutter Unit to millimeter?

Thanks and regards,
Viet

**Comment by iapicca** [comment]
@tixngo-mobile-github 
I believe at the moment flutter doesn't expose the size of screen in `mm` or `inches`
(I'd be gladly proven wrong)
I really would love to have access to such information, although this would probably imply a native plugin
rather than a direct implementation in the sdk
(again... I'd be gladly proven wrong)

imho this is both a valid [bug on wrong doc](https://stackoverflow.com/a/50647601/9836706) and a useful `feature request`

**Comment by TahaTesser** [comment]
I don't think it is possible to exact screen size from Flutter itself hence correcting the issue for doc

**Comment by nt4f04uNd** [comment]
The https://api.flutter.dev/flutter/dart-ui/FlutterView/devicePixelRatio.html says

> The number of device pixels for each logical pixel for the screen this view is displayed on.

According to that, to convert logical pixels to physical pixels you would need to multiply by this value. What piece of documentation you are referring to when you say that it needs correction?

**Comment by nt4f04uNd** [comment]
Reading further I'd personally change/remove this part

> By definition, there are roughly 38 logical pixels per centimeter, or about 96 logical pixels per inch, of the physical display. The value returned by devicePixelRatio is ultimately obtained either from the hardware itself, the device drivers, or a hard-coded value stored in the operating system or firmware, and may be inaccurate, sometimes by a significant margin.

By what definition? Why 38 if that is variable? That is very misleading

**Comment by esDotDev** [comment]
At the very least the docs can be quickly updated to say that it is 96px on desktop / web, but more like 160px on mobile devices. The just leave the 96 in there is really confusing. 

But as noted in the OP, sometimes the `devicePixelRatio` just gets way off. For example it reports 2.0 on iPad Pro, which gives a computer DPI of 320, but the actual DPI of iPad Pro is 265. This is WAY off and limits our ability to adapt to physical screen size. 

iPad Mini and iPad tend to calculate close to the same physical size, when clearly they are not. 

Why is this the case? I understand for Android sometimes the drivers are unreliable and there are too many devices to account for, but seems like the iOS devices could be hard-coded easily enough to their proper values.
